src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -65,22 +66,32 @@ public async Task<IActionResult> Get(string name)
         [HttpPut]
         [Route("admin/functions/{name}")]
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
-        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata)
+        public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata, [FromServices] IEnumerable<IHostedService> hostedServices)
         {
-            if (!Utility.IsValidFunctionName(name))
+            FileMonitoringService fileMonitoringService = (FileMonitoringService)hostedServices.SingleOrDefault(x => x is FileMonitoringService);
```

Comment:
This will never be null, right? Assuming so, rather than doing a manual lookup like this, can't you just take FileMonitoringService as a ctor DI param?

Reply:
we still need FileMonitoringService added as IHostedService to be a background task. See the comment above.

Reply:
Comment where? The service would be registered as both, a hosted service, for background processing, as well as a file monitoring service.